ragchain-structure/benchmark/dataset-evaluator/beir.md

```diff
@@ -0,0 +1,63 @@
+# Overview
+
+BEIR is a heterogeneous benchmark that has been built from 18 diverse datasets representing 9 information retrieval tasks.
+Using project, we use 6 datasets, FEVER, FIQA, HOTPOTQA, QUORA, SCIDOCS, and SCIFACT.
+More information about beir dataset, please visit link!
+
+[Github](https://github.com/beir-cellar/beir?tab=readme-ov-file)<br>
+[Paper](https://openreview.net/forum?id=wCu6T5xFjeJ)<br>
+[HuggningFace](https://huggingface.co/BeIR)
+
+
+Our BEIR benchmark structure is like below picture.
+Basic preprocessing process is same at 6 benchmark dataset,
+So we construct preprocess process, ingest, evaluation code are in BaseBeirEvaluator class.
+Each evaluator class corresponding to the benchmark datasets
+(FEVER, FIQA, HOTPOTQA, QUORA, SCIDOCS, and SCIFACT) inherits from the BaseBeirEvaluator.
+
+![beir-structure.png](../../../.gitbook/assets/beir-structure.png)
+
+Also, we removed non-relevant score passages at qrels.<br>
+The significance of this dataset is diverse tasks, diverse domains, task difficulties, and diverse annotation strategies
+in each dataset.
+
+Beir test code is little different from other benchmark.
+```Python
+bm25_path, pickle_path = generate_path('fever')
+retrieval, db, llm, pipeline = set_beir_evaluator(bm25_path, pickle_path)
```

Comment:
also, where is set_beir_evaluator?
Don't just copy and paste pytest fixture codes. Write adequate code for newbies.